### ISSUE TITLE
Implement local bookmarking

### ISSUE THREAD

**Issue by aferditamuriqi** [NYPL-Simplified/Simplified-iOS#605]
this should work on device only, server sync component will be addressed in a separate issue

referencing https://github.com/NYPL-Simplified/Simplified/issues/23

**Comment by aferditamuriqi** [comment]
Page numbering needs to be addressed still. currently showing Content CFI.

Chapter title also needs some refinement, as of now if there is no chapter name, we need to figure out what information to display.

Excerpt has been left out for now, but maybe that is something which can be looked at when chapter name is being addressed.

**Comment by aferditamuriqi** [comment]
There seems to be one issue, which needs testing and steps to reproduce:

when a book is downloaded the first time, bookmarks seem to not  persist, only after relaunching the book then bookmarks seems to works as expected.

**Comment by aferditamuriqi** [comment]
closing this with PR https://github.com/NYPL-Simplified/Simplified-iOS/pull/624
